### ISSUE TITLE
KV key-name case mismatch

### ISSUE THREAD

**Issue by aantono** [containous/traefik#2103]
### Do you want to request a *feature* or report a *bug*?
Discovered a BUG in using `storeconfig` and later reading data using `kv` provider (`consul`).
 It appears that staert/kv library (https://github.com/containous/traefik/blob/master/vendor/github.com/containous/staert/kv.go#L233) always converts the key to lowercase.
The template for kv/kv provider (https://github.com/containous/traefik/blob/master/templates/kv.tmpl#L51) uses proper casing, as it is defined in the golang structs.

### What did you do?

Executed `./traefik --configFile=configs/traefik_dev.toml --debug --consul --consul.endpoint="consul.***:8500"`

### What did you expect to see?
Expected to get correct value for `passHostHeader` setting (equal to false)

### What did you see instead?
Got an error in the log saying `Cannot get key traefik/frontends/na-nginx/passHostHeader Key not found in store, setting default true`

### Output of `traefik version`: (_What version of Traefik are you using?_)
```
Version:      v1.4.0-rc2
Codename:     roquefort
Go version:   go1.9
Built:        2017-09-08_07:31:37PM
OS/Arch:      darwin/amd64
```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```toml
################################################################
# Global configuration
################################################################

logLevel = "DEBUG"

defaultEntryPoints = ["http"]

MaxIdleConnsPerHost = 200 # Default: 200

IdleTimeout = "180s" # Default: 180s

InsecureSkipVerify = true # Default false

[healthcheck]
interval = "30s" # Defautl 30s

################################################################
# Entrypoints
################################################################

[entryPoints]
  [entryPoints.http]
  address = ":8081"

################################################################
# API Backend
################################################################

[web]
  address = ":8080"

################################################################
# Consul KV Backend
################################################################

[consul]
endpoint = "consul.***:8500"
watch = true
prefix = "traefik"
```

### If applicable, please paste the log output in debug mode (`--debug` switch)

```
DEBU[2017-09-11T15:31:18-05:00] Cannot get key traefik/backends/na-nginx-lab/circuitbreaker/expression Key not found in store, setting default  
DEBU[2017-09-11T15:31:18-05:00] Cannot get key traefik/backends/na-nginx-lab/loadbalancer/method Key not found in store, setting default  
DEBU[2017-09-11T15:31:18-05:00] Cannot get key traefik/backends/na-nginx-lab/loadbalancer/sticky Key not found in store, setting default false 
DEBU[2017-09-11T15:31:18-05:00] Cannot get key traefik/backends/na-nginx-lab/healthcheck/path Key not found in store, setting default  
DEBU[2017-09-11T15:31:18-05:00] Cannot get key traefik/backends/na-nginx-lab/circuitbreaker/expression Key not found in store, setting default  
DEBU[2017-09-11T15:31:18-05:00] Cannot get key traefik/backends/na-nginx-lab/maxconn/amount Key not found in store, setting default  
DEBU[2017-09-11T15:31:18-05:00] Cannot get key traefik/backends/na-nginx-lab/loadbalancer/method Key not found in store, setting default  
WARN[2017-09-11T15:31:18-05:00] Error checking new version: Malformed version: d54417acfe2be53021c60576e645964dc2d3a236 
DEBU[2017-09-11T15:31:19-05:00] Cannot get key traefik/backends/na-nginx-lab/maxconn/extractorfunc Key not found in store, setting default  
DEBU[2017-09-11T15:31:19-05:00] Cannot get key traefik/backends/na-nginx-lab/loadbalancer/sticky Key not found in store, setting default false 
DEBU[2017-09-11T15:31:19-05:00] Cannot get key traefik/backends/na-nginx-lab/healthcheck/path Key not found in store, setting default  
DEBU[2017-09-11T15:31:19-05:00] Cannot get key traefik/backends/na-nginx-lab/maxconn/amount Key not found in store, setting default  
DEBU[2017-09-11T15:31:19-05:00] Cannot get key traefik/backends/na-nginx-lab/maxconn/extractorfunc Key not found in store, setting default  
DEBU[2017-09-11T15:31:19-05:00] Cannot get key traefik/backends/traveler-risk-management-beta/circuitbreaker/expression Key not found in store, setting default  
DEBU[2017-09-11T15:31:19-05:00] Cannot get key traefik/backends/traveler-risk-management-beta/loadbalancer/method Key not found in store, setting default  
DEBU[2017-09-11T15:31:19-05:00] Cannot get key traefik/backends/traveler-risk-management-beta/loadbalancer/sticky Key not found in store, setting default false 
DEBU[2017-09-11T15:31:20-05:00] Cannot get key traefik/backends/traveler-risk-management-beta/healthcheck/path Key not found in store, setting default  
DEBU[2017-09-11T15:31:20-05:00] Cannot get key traefik/backends/traveler-risk-management-beta/circuitbreaker/expression Key not found in store, setting default  
DEBU[2017-09-11T15:31:20-05:00] Cannot get key traefik/backends/traveler-risk-management-beta/maxconn/amount Key not found in store, setting default  
DEBU[2017-09-11T15:31:20-05:00] Cannot get key traefik/backends/traveler-risk-management-beta/loadbalancer/method Key not found in store, setting default  
DEBU[2017-09-11T15:31:20-05:00] Cannot get key traefik/backends/traveler-risk-management-beta/maxconn/extractorfunc Key not found in store, setting default  
DEBU[2017-09-11T15:31:20-05:00] Cannot get key traefik/backends/traveler-risk-management-beta/loadbalancer/sticky Key not found in store, setting default false 
DEBU[2017-09-11T15:31:20-05:00] Cannot get key traefik/backends/traveler-risk-management-beta/healthcheck/path Key not found in store, setting default  
DEBU[2017-09-11T15:31:20-05:00] Cannot get key traefik/backends/traveler-risk-management-beta/maxconn/amount Key not found in store, setting default  
DEBU[2017-09-11T15:31:20-05:00] Cannot get key traefik/backends/traveler-risk-management-beta/maxconn/extractorfunc Key not found in store, setting default  
DEBU[2017-09-11T15:31:20-05:00] Cannot get key traefik/backends/traveler-risk-management-service-beta/circuitbreaker/expression Key not found in store, setting default  
DEBU[2017-09-11T15:31:20-05:00] Cannot get key traefik/backends/traveler-risk-management-service-beta/loadbalancer/method Key not found in store, setting default  
DEBU[2017-09-11T15:31:21-05:00] Cannot get key traefik/backends/traveler-risk-management-service-beta/loadbalancer/sticky Key not found in store, setting default false 
DEBU[2017-09-11T15:31:21-05:00] Cannot get key traefik/backends/traveler-risk-management-service-beta/healthcheck/path Key not found in store, setting default  
DEBU[2017-09-11T15:31:21-05:00] Cannot get key traefik/backends/traveler-risk-management-service-beta/circuitbreaker/expression Key not found in store, setting default  
DEBU[2017-09-11T15:31:21-05:00] Cannot get key traefik/backends/traveler-risk-management-service-beta/maxconn/amount Key not found in store, setting default  
DEBU[2017-09-11T15:31:21-05:00] Cannot get key traefik/backends/traveler-risk-management-service-beta/loadbalancer/method Key not found in store, setting default  
DEBU[2017-09-11T15:31:21-05:00] Cannot get key traefik/backends/traveler-risk-management-service-beta/maxconn/extractorfunc Key not found in store, setting default  
DEBU[2017-09-11T15:31:21-05:00] Cannot get key traefik/backends/traveler-risk-management-service-beta/loadbalancer/sticky Key not found in store, setting default false 
DEBU[2017-09-11T15:31:21-05:00] Cannot get key traefik/backends/traveler-risk-management-service-beta/healthcheck/path Key not found in store, setting default  
DEBU[2017-09-11T15:31:21-05:00] Cannot get key traefik/frontends/na-nginx/entrypoints Key not found in store, setting default empty 
DEBU[2017-09-11T15:31:21-05:00] Cannot get key traefik/backends/traveler-risk-management-service-beta/maxconn/amount Key not found in store, setting default  
DEBU[2017-09-11T15:31:21-05:00] Cannot get key traefik/backends/traveler-risk-management-service-beta/maxconn/extractorfunc Key not found in store, setting default  
DEBU[2017-09-11T15:31:21-05:00] Cannot get key traefik/frontends/na-nginx/passHostHeader Key not found in store, setting default true 
DEBU[2017-09-11T15:31:22-05:00] Cannot get key traefik/frontends/na-nginx/entrypoints Key not found in store, setting default empty 
DEBU[2017-09-11T15:31:22-05:00] Cannot get key traefik/frontends/travel-risk-management-service/entrypoints Key not found in store, setting default empty 
DEBU[2017-09-11T15:31:22-05:00] Cannot get key traefik/frontends/na-nginx/passHostHeader Key not found in store, setting default true 
DEBU[2017-09-11T15:31:22-05:00] Cannot get key traefik/frontends/travel-risk-management-service/passHostHeader Key not found in store, setting default true 
DEBU[2017-09-11T15:31:22-05:00] Cannot get key traefik/frontends/travel-risk-management-service/entrypoints Key not found in store, setting default empty 
DEBU[2017-09-11T15:31:22-05:00] Cannot get key traefik/frontends/travel-risk-management/entrypoints Key not found in store, setting default empty 
DEBU[2017-09-11T15:31:22-05:00] Cannot get key traefik/frontends/travel-risk-management-service/passHostHeader Key not found in store, setting default true 
DEBU[2017-09-11T15:31:23-05:00] Cannot get key traefik/frontends/travel-risk-management/passHostHeader Key not found in store, setting default true 
DEBU[2017-09-11T15:31:23-05:00] Cannot get key traefik/frontends/travel-risk-management/entrypoints Key not found in store, setting default empty 
DEBU[2017-09-11T15:31:23-05:00] Validation of load balancer method for backend na-nginx-lab failed: invalid load-balancing method ''. Using default method wrr. 
```

**Comment by aantono** [comment]
It seems like the fix could be to convert all the keys to lower-case in https://github.com/containous/traefik/blob/master/provider/kv/kv.go.  Thoughts?

**Comment by traefiker** [comment]
Closed by #3015.
